### ISSUE TITLE
Training Mechanism

### ISSUE THREAD

**Issue by Yacovitch** [loicland/superpoint_graph#137]
Hi,

In the paper, GRU's initial hidden state is a vector from embedding.
I am wondering whether PointNet and GRU are trained separately or together.
If it is trained separately, what labels do you use to train the embedding algorithm with PointNet.
If it is trained together, how do you back propagate all the way up to PointNet?

Thanks

**Comment by loicland** [comment]
PointNet and GRU are trained together in an end-to-end fashion.

The gradients backpropagate through the 10 GCN iterations to the initial state, and to the PointNets.
